### PR TITLE
Ability to delete a Value Data Type

### LINKED ISSUE #242: add trash can delete for Value Data Type
When you delete a URI from the Value Data Type field in the Profile editor, the field does not go away. When you look at the JSON, the field data is still there with an empty value. This causes an error message when you try to load the template into the Linked Data editor.
JSON

### PR DESCRIPTION
**Pull request by jgreben**
Fixes #242 
- Link to add Value Data Type instead of the field existing by default
- Sets valueDataType to empty value by default; must initialized to an array in order to display it as a valueConstraint list item, but value is still rendered as a single object.
- Checks if there is a value in valueDataType and hides the Add link to ensure there is only one value that can be added.
- Removes `"dataTypeLabel"`, `"dataTypeLabelHint"`, and `"remark"` from the rendered valueDataType json.
- Tests for adding and removing the field from the view

VDT

Filled in value:
```
{
  "Profile": {
    "resourceTemplates": [
      {
        "propertyTemplates": [
          {
            "mandatory": "false",
            "repeatable": "true",
            "type": "literal",
            "resourceTemplates": [],
            "valueConstraint": {
              "valueTemplateRefs": [],
              "useValuesFrom": [],
              "valueDataType": {
                "dataTypeURI": "http://example.com#test"
              },
              "defaults": []
            },
            "propertyURI": "http://ex.com/test",
            "propertyLabel": "test"
          }
        ],
        "id": "test",
        "resourceURI": "http://ex.com",
        "resourceLabel": "test",
        "author": "test",
        "schema": "https://ld4p.github.io/sinopia/schemas/0.0.9/resource-template.json"
      }
    ],
    "id": "profile.test",
    "description": "test",
    "author": "test",
    "title": "test",
    "schema": "https://ld4p.github.io/sinopia/schemas/0.0.9/profile.json",
    "date": "2019-11-06"
  }
}
```

Removed value:
```
{
  "Profile": {
    "resourceTemplates": [
      {
        "propertyTemplates": [
          {
            "mandatory": "false",
            "repeatable": "true",
            "type": "literal",
            "resourceTemplates": [],
            "valueConstraint": {
              "valueTemplateRefs": [],
              "useValuesFrom": [],
              "defaults": []
            },
            "propertyURI": "http://ex.com/test",
            "propertyLabel": "test"
          }
        ],
        "id": "test",
        "resourceURI": "http://ex.com",
        "resourceLabel": "test",
        "author": "test",
        "schema": "https://ld4p.github.io/sinopia/schemas/0.0.9/resource-template.json"
      }
    ],
    "id": "profile.test",
    "description": "test",
    "author": "test",
    "title": "test",
    "schema": "https://ld4p.github.io/sinopia/schemas/0.0.9/profile.json",
    "date": "2019-11-06"
  }
}
```

### CODE REVIEW

**Comment by jgreben** [comment]
Profile does not validate if valueDataType is an empty array.

**Comment by jgreben** [comment]
Fixed profile validation by removing the initialized valueDataType array if it is empty.  If it has a value, it is already rendered as an object.

**Comment by jgreben** [comment]
Need to add tests for generated profile on export.

**Comment by jgreben** [comment]
Added test for correctly generated profile json.

File: source/assets/js/modules/profile/services/vocab.service.js [hunk 1]
```diff
@@ -415,8 +415,8 @@ angular.module('locApp.modules.profile.services').factory('Vocab', function($q,
     fillDatatype = function(vocab) {
         var datatype = choosenDatatype;
         datatype.dataTypeURI = vocab.uri;
-        datatype.dataTypeLabel = vocab.label;
-        datatype.remark = vocab.comment;
+        // datatype.dataTypeLabel = vocab.label;
```
Comment (justinlittman) [inline source/assets/js/modules/profile/services/vocab.service.js]: Cruft

File: source/assets/js/modules/profile/services/profileHandler.service.js [hunk 2]
```diff
@@ -15,7 +15,8 @@ angular.module('locApp.modules.profile.services')
         var resAttributes = ["id","resourceURI","resourceLabel","propertyTemplates","author","date","remark","schema","adherence","source"];
         var propAttributes = ["propertyURI","propertyLabel","mandatory","repeatable","type","valueConstraint","remark", "resourceTemplates"];
         var consAttributes = ["valueLanguage","languageURI","languageLabel","valueDataType","valueTemplateRefs","useValuesFrom","editable","remark", "repeatable", "defaultURI", "defaultLiteral", "defaults", "validatePattern"];
-        var dataAttributes = ["dataTypeURI","dataTypeLabel","dataTypeLabelHint","remark"];
+        // var dataAttributes = ["dataTypeURI","dataTypeLabel","dataTypeLabelHint","remark"];
```
Comment (justinlittman) [inline source/assets/js/modules/profile/services/profileHandler.service.js]: Cruft

File: source/html/valueConstraint.html [hunk 3]
```diff
@@ -35,10 +35,20 @@ <h5 id="constraintHeader">Value Constraint</h5>
...
-        <div ng-repeat="field in constraintFields" sss-field item="{{field}}" html='html/valueDataType.html'></div>
+        <h5>Value Data Type</h5>
+        <div class="listItems" ng-repeat="field in valueDataType" sss-field item="{{field}}" html='html/valueDataType.html'></div>
+        <a id="addValueDataType" ng-show="!valueDataType.length" ng-click="addValueDataType()" style="display: block;">
+            <i class="fa fa-plus"></i>
+            Add Value Data Type
+        </a>
     </div>
 
+    <!--<div id="valueDataType" class="propertyItems">-->
```
Comment (justinlittman) [inline source/html/valueConstraint.html]: Cruft

File: source/html/valueDataType.html [hunk 4]
```diff
@@ -1,22 +1,27 @@
-<div ng-controller="ValueDataTypeController">
-    <h5>Value Data Type</h5>
+<div id="{{item}}_{{parentId}}" ng-controller="ValueDataTypeController">
+    <!--<h5>Value Data Type</h5>-->
```
Comment (justinlittman) [inline source/html/valueDataType.html]: Cruft

**Pull request merged**